### PR TITLE
Feat(modules-sdk): Run hooks on application start

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- run `onApplicationStart` hook when starting modules

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

one of many missing pieces from [that PR](https://github.com/medusajs/medusa/pull/5447/files) hehe

File: packages/modules-sdk/src/medusa-module.ts [hunk 1]
```diff
@@ -85,6 +86,22 @@ export class MedusaModule {
       return MedusaModule.getModuleInstance(key)
     })
   }
+  
+  public static onApplicationStart(): void {
+    for (const instances of MedusaModule.instances_.values()) {
+      for (const instance of Object.values(instances) as IModuleService[]) {
+        console.log(instance)
```
Comment (carlos-r-l-rodrigues) [inline packages/modules-sdk/src/medusa-module.ts]: rm

**Review by adrien2p — APPROVED** [review]
LGTM, exactly 😄 but I thought phil could do it since it will be needed probably soon ahah

File: packages/modules-sdk/src/medusa-module.ts (hunk 1 above)
Comment (pKorsholm) [reply packages/modules-sdk/src/medusa-module.ts]: 😅

**Pull request merged**